doc/usage/configuration.rst

```diff
@@ -2522,6 +2522,25 @@ Options for the linkcheck builder
 
    .. versionadded:: 1.1
 
+.. confval:: linkcheck_ignore_redirects
+
+   A dictionary that maps a pattern of the source URI to a pattern of the canonical
+   URI.  If set, linkcheck builder treated as "ignored" if a hyperlink in the source
+   document that matches to the pattern of the source URI, and it redirects to the
+   URL that matches to the pattern of the canonical URI.  Example:
+
+   .. code-block:: python
+
+      linkcheck_ignore_redirects = {
+          # All HTTP redirections from the source URI to the canonical URI will be ignored.
+          'http://sphinx-doc.org/.*': 'https://sphinx-doc.org/en/master/.*'
```

Comment:
I don't think the example `'https://sphinx-doc.org/.*': 'https://sphinx-doc.org/en/master/.*'` is good because it allows any redirect without checking that the page is actually the same.

For the specific case of `https://sphinx-doc.org/`, a redirect to anything in `https://(www\.)?sphinx-doc.org/.*` would be ok because the intent is just to link to the root page of the website.

Reply:
Since these are regexps it would have to be `r'https://sphinx-doc\.org/.*': r'https://sphinx-doc\.org/en/master/.*'` too...